### PR TITLE
feat: support network proxy settings

### PR DESCRIPTION
**Pull request by hui-yang**
**Issue #, if available:**

**Description of changes:**

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java [hunk 1]
```diff
@@ -95,8 +94,9 @@ private void mockDefaultGetGroups() throws ServiceLoadException {
     }
 
     @Test
-    void GIVEN_one_system_component_one_file_less_than_max_WHEN_merge_THEN_reads_entire_file()
+    void GIVEN_one_system_component_one_file_less_than_max_WHEN_merge_THEN_reads_entire_file(ExtensionContext ec)
             throws URISyntaxException, ServiceLoadException {
+        ignoreExceptionOfType(ec, DateTimeParseException.class);
```
Comment (nikkhilmuthye) [inline src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java:99]: Why do we need to ignore this exception?

Comment (hui-yang) [reply src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java:99]: the exception's logged here https://github.com/aws-greengrass/aws-greengrass-log-manager/blob/14c9c769f74e0dd934f54522bd75c926fcfe0ce0/src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java#L259-L262

File: src/main/java/com/aws/greengrass/logmanager/CloudWatchLogsUploader.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ private boolean uploadLogs(String logGroupName, String logStreamName, List<Input
             return true;
         } catch (ResourceNotFoundException e) {
             // Handle no log group/log stream
-            logger.atError().cause(e).log("Unable to find log group- {} or log stream - {}. Creating them now.",
+            logger.atInfo().cause(e).log("Unable to find log group- {} or log stream - {}. Creating them now.",
```
Comment (nikkhilmuthye) [inline src/main/java/com/aws/greengrass/logmanager/CloudWatchLogsUploader.java:146]: We should probably update this log to say that we will be creating the log group/stream.

File: src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java (hunk 1 above)
Comment (nikkhilmuthye) [reply src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java:99]: Ok. I was just confused that this should have been caught when that change was made.

**Review by nikkhilmuthye — APPROVED** [review]

**Pull request merged**